src/services/user-config-service.ts

```diff
@@ -116,12 +116,7 @@ const convertYamlToUserConfig = (input?: string): Config => {
 	const deployments = {};
 	if (configDeployments != null) {
 		if (configDeployments.environmentMapping) {
-			deployments["environmentMapping"] =  {
-				development: configDeployments.environmentMapping.development,
-				testing: configDeployments.environmentMapping.testing,
-				staging: configDeployments.environmentMapping.staging,
-				production: configDeployments.environmentMapping.production
-			};
+			deployments["environmentMapping"] = configDeployments.environmentMapping;
```

Comment:
If the lack of key filtering is concerning here, I could change this to:
```js
deployments["environmentMapping"] = pickBy(configDeployments.environmentMapping, (_, key) => ['development','testing','staging','production'].includes(key));
```

Lodash `pick` doesn't maintain key order, but `pickBy` does

Reply:
My thinking is, if we filter by key names, and user has a config like
```
{
  prod: ["prod-*"],
  development: ["*"],
}
```
and has a deployment with env name `prod-us-west-1`, the app will map it to `development` as the previous key `prod` will be omitted by the `pickBy`. Which leads to confusion to users.

However, without the filter by key, apps will store `prod` as the key in db, but `prod` is not supported in the upstream  jira api that save the deployment data 
api: https://developer.atlassian.com/cloud/jira/software/rest/api-group-deployments/#api-rest-deployments-0-1-bulk-post
![image](https://github.com/atlassian/github-for-jira/assets/105693507/2e8ca0b5-6424-4eff-8334-6afa7b3e8fe1)


I suggest
1. We don't do key filter here. store whatever user supplied in yaml file in db
2. However, during mapping, map it to `unmapped` when that key is not in the supported array found in api doc.

Reply:
And I mean checking whether the final mapping is supported and map to `unmapped` if not supported.

here: https://github.com/atlassian/github-for-jira/blob/main/src/transforms/transform-deployment.ts#L222